src/Language/PureScript/Scheme/CodeGen/Transpiler.hs

```diff
@@ -140,8 +161,23 @@ caseToScheme values caseAlternatives =
     -- We have to replace each occurrence of `m' with `v' and each occurrence
     -- of `n' with `v0'.
     variablesToReplace valuesAndBinders =
-      [ (runIdent ident, exprToScheme value)
-      | (value, (VarBinder _ann ident)) <- valuesAndBinders ]
+      concatMap go valuesAndBinders
+      where
+        go (_value, NullBinder _ann) = []
+        go (_value, LiteralBinder _ann _literal) = []
+        go (value, VarBinder _ann ident) = [(runIdent ident, exprToScheme value)]
+        go (value, ConstructorBinder _ann _typeName _constructorName binders) =
+          map (\(i, b) -> (go' b, vectorRef (exprToScheme value)
+                                            (IntegerLiteral i)))
+              (zip [1 .. (genericLength binders)]
+                   binders)
+        go _ = error "Not implemented"
```

Comment:
Can we explicitly match on all cases?

Reply:
So like `var@(Bar i1 i2)`? If that's so, can you add this to the test? (also referencing `var` in the body so we make sure to handle the renaming)

Reply:
The wollowing test case

```
data Foo = Bar Int Int
         | Baz

match var@(Bar i1 i2) = var
match Baz = Baz
```

hits `binderToTest` with a NamedBinder. I'd rather implement it in the next PR since there is already a lot going on in this one. Let me know if I have to test with a different case though.

Reply:
Yep, this test looks good, and fixing this in the next PR is a good idea too

Reply:
Issue #29 will take care of this